src/components/dialogs/parameters/dynamicsimulation/curve/dialog/equipment-filter.js

```diff
@@ -46,253 +47,172 @@ const styles = {
-        useEffect(() => {
-            Promise.all(
-                EQUIPMENT_FETCHERS.VOLTAGE_LEVEL.map((fetchPromise) =>
-                    fetchPromise(studyUuid, currentNode.id)
-                )
-            )
-                .then((vals) => {
-                    // convert array to voltageLevels object
-                    const voltageLevelsObj = vals
-                        .flat()
-                        .reduce(
-                            (obj, curr) => ({ ...obj, [curr.id]: curr.name }),
-                            {}
-                        );
-
-                    // convert array to tensions object
-                    const tensionObj = vals.flat().reduce(
-                        (obj, curr) => ({
-                            ...obj,
-                            [`${curr.nominalVoltage}`]: curr.nominalVoltage,
-                        }),
-                        {}
-                    );
-
-                    // update voltage level states
-                    setVoltageLevels(voltageLevelsObj);
-                    setSelectedVoltageLevelIds(Object.keys(voltageLevelsObj));
-
-                    // update tension states
-                    setTensions(tensionObj);
-                    setSelectedTensionIds(Object.keys(tensionObj));
-
-                    // update loading state
-                    setSubstationsFiltersReady(true);
-                })
-                .catch((error) => {
-                    snackError({
-                        messageTxt: error.message,
-                        headerId: 'DynamicSimulationFetchVoltageLevelError',
-                    });
-                });
-        }, [currentNode.id, studyUuid, snackError]);
-
-        // --- country (countryCode metadata) => lookup in substation --- //
-        const [selectedCountries, setSelectedCountries] = useState([]);
+        // --- country (i.e. countryCode) => fetch from network-map-server --- //
         const [countries, setCountries] = useState([]);
+        const [selectedCountries, setSelectedCountries] = useState([]);
         const handleCountryChange = useCallback((selectedCountries) => {
             setSelectedCountries(selectedCountries);
         }, []);
 
-        // load substation from backend to infer countries
+        // load countries
         useEffect(() => {
-            Promise.all(
-                EQUIPMENT_FETCHERS.SUBSTATION.map((fetchPromise) =>
-                    fetchPromise(studyUuid, currentNode.id)
-                )
-            )
-                .then((vals) => {
-                    // get countries codes
-                    let countries = [
-                        ...vals.flat().reduce((set, substation) => {
-                            substation.countryCode &&
-                                set.add(substation.countryCode);
-                            return set;
-                        }, new Set()),
-                    ];
-
+            fetchAllCountries(studyUuid, currentNode.id)
```

Comment:
When node is invalid we got en error

Reply:
Done : https://github.com/gridsuite/gridstudy-app/commit/fef4bab75d83dc7ef8bd57d92ba9343d123b6c51#diff-004e2fd93b0e13e660ba95c75ebe656a464ba5f4810c28fd61ce1feb353ceaf6R24